src/lib/db.js

```diff
@@ -4,35 +4,64 @@ import _ from 'lodash';
 
 let _db = undefined;
 
-export class Activity {
+// In the future, use private fields instead:
+//   https://github.com/tc39/proposal-class-fields
+// I thought I could use symbols, but that didn't work out.
+let modelAttrs = {};
+
+class Model {
+  constructor(table, key) {
+    modelAttrs[this.constructor] = { table: table, key: key };
+  }
+
+  async save() {
+    // Does an update if the row already exists, otherwise does a put.
+    let keyname = modelAttrs[this.constructor].key;
+    let key = this[keyname];
+    let table = modelAttrs[this.constructor].table;
+    const exists =
```

Comment:
Couldn't this just be
``` javascript
const exists = await table.get(this[keyname])
```
which becomes `undefined` when the primary key `this[keyname]` isn't in the store?

Couldn't we also skip all of it and do:
``` js
return table.add(this).catch(err => table.update(key, this))
```
? (also avoids race conditions, however that's not a major concern)

I'm fine either way, just for your consideration.

Reply:
Ah, nice. Fixed.